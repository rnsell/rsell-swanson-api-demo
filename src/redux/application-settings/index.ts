export * from "./application-settings.reducer";
