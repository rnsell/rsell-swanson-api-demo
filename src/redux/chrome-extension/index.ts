export * from "./chrome-extension.actions";
export * from "./chrome-extension.epic";
