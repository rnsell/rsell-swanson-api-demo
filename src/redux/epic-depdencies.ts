import { getRonSwonsonQuotes } from "../services";

export type EpicDependencies = {
  getRonSwonsonQuotes: (numberofQuotes?: number) => Promise<string[]>;
};

export const generateEpicDependencies = (): EpicDependencies => {
  return {
    getRonSwonsonQuotes: getRonSwonsonQuotes(),
  };
};
