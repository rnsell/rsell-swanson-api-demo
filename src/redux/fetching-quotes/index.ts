export * from "./fetching-quotes.actions";
export * from "./fetching-quotes.epic";
export * from "./fetching-quotes.model";
export * from "./fetching-quotes.reducer";
