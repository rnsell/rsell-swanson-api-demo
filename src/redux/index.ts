export * from "./root-reducer";
export * from "./root-epic";
export * from "./create-store";
export * from "./theme";
export * from "./application-settings";
export * from "./fetching-quotes";
export * from "./quotes-table";
export * from "./chrome-extension";
