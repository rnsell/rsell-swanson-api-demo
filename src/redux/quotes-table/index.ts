export * from "./quotes-table.actions";
export * from "./quotes-table.model";
export * from "./quotes-table.reducers";
