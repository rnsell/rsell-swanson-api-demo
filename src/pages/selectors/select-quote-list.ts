import { downVote, Quote, upVote } from "../../redux";
import { QuoteProps } from "../../components";
import { Action } from "redux";

type DispatchFunction = (action: Action) => void;

export const selectQuoteProps = (dispatch: DispatchFunction) => (
  quote: Quote
): QuoteProps => {
  const { id } = quote;

  const onUpVoteClick = () => {
    const upVoateAction = upVote(id);
    dispatch(upVoateAction);
  };

  const onDownVoteClick = () => {
    const downVoteAction = downVote(id);
    dispatch(downVoteAction);
  };

  const quoteProps: QuoteProps = {
    ...quote,
    onUpVoteClick,
    onDownVoteClick,
  };

  return quoteProps;
};

export const generateListOfQuotes = (
  dispath: DispatchFunction,
  quotes: Quote[]
): QuoteProps[] => {
  return quotes.map(selectQuoteProps(dispath));
};
