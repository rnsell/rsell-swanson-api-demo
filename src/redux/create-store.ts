import { createStore, applyMiddleware } from "redux";
import { composeWithDevTools } from "redux-devtools-extension";
import { createEpicMiddleware } from "redux-observable";
import { rootReducer } from "./root-reducer";
import { rootEpic } from "./root-epic";
import { generateEpicDependencies } from "./epic-depdencies";

const epicMiddleware = createEpicMiddleware({
  dependencies: generateEpicDependencies(),
});

export const configureStore = (baseState: any = {}) => {
  const store = createStore(
    rootReducer,
    baseState,
    composeWithDevTools(applyMiddleware(epicMiddleware))
  );

  epicMiddleware.run(rootEpic);

  return store;
};
