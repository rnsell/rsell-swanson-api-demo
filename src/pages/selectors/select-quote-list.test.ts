import { Quote } from "../../redux";
import { generateListOfQuotes, selectQuoteProps } from "./select-quote-list";

describe("select-quote-list.ts", () => {
  const aRandomQuote = new Quote({
    quoteText: "Go Beyond Plus Ultra!!",
    votes: 9000,
    upVoteSelected: false,
    downVoteSelected: false,
  });

  describe("selectQuoteProps", () => {
    it("Should dispatch actions", () => {
      const dispatch = jest.fn(() => void 0);

      const quoteProp = selectQuoteProps(dispatch)(aRandomQuote);

      expect(quoteProp).toMatchSnapshot();
      quoteProp.onUpVoteClick();
      quoteProp.onDownVoteClick();

      expect(dispatch).toHaveBeenCalledTimes(2);

      const [upVoteCall, downVoteCall] = dispatch.mock.calls;
      expect(upVoteCall).toMatchSnapshot();
      expect(downVoteCall).toMatchSnapshot();
    });
  });

  it("Should generate list of quotes", () => {
    const dispatch = jest.fn(() => void 0);
    const list = generateListOfQuotes(dispatch, [aRandomQuote]);

    expect(list).toMatchSnapshot();
  });
});
