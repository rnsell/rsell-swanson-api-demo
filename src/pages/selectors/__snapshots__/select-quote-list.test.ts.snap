// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`select-quote-list.ts Should dispatch actions 1`] = `
Object {
  "downVoteSelected": false,
  "id": "a936264fd5671771d4033893764022d2",
  "onDownVoteClick": [Function],
  "onUpVoteClick": [Function],
  "quoteText": "Go Beyond Plus Ultra!!",
  "upVoteSelected": false,
  "votes": 9000,
}
`;

exports[`select-quote-list.ts Should dispatch actions 2`] = `
Array [
  Object {
    "payload": Object {
      "clickType": "UP_VOTE",
      "quoteId": "a936264fd5671771d4033893764022d2",
    },
    "type": "QUOTES_TABLE::CLICK",
  },
]
`;

exports[`select-quote-list.ts Should dispatch actions 3`] = `
Array [
  Object {
    "payload": Object {
      "clickType": "DOWN_VOTE",
      "quoteId": "a936264fd5671771d4033893764022d2",
    },
    "type": "QUOTES_TABLE::CLICK",
  },
]
`;

exports[`select-quote-list.ts Should generate list of quotes 1`] = `
Array [
  Object {
    "downVoteSelected": false,
    "id": "a936264fd5671771d4033893764022d2",
    "onDownVoteClick": [Function],
    "onUpVoteClick": [Function],
    "quoteText": "Go Beyond Plus Ultra!!",
    "upVoteSelected": false,
    "votes": 9000,
  },
]
`;

exports[`select-quote-list.ts selectQuoteProps Should dispatch actions 1`] = `
Object {
  "downVoteSelected": false,
  "id": "a936264fd5671771d4033893764022d2",
  "onDownVoteClick": [Function],
  "onUpVoteClick": [Function],
  "quoteText": "Go Beyond Plus Ultra!!",
  "upVoteSelected": false,
  "votes": 9000,
}
`;

exports[`select-quote-list.ts selectQuoteProps Should dispatch actions 2`] = `
Array [
  Object {
    "payload": Object {
      "clickType": "UP_VOTE",
      "quoteId": "a936264fd5671771d4033893764022d2",
    },
    "type": "QUOTES_TABLE::CLICK",
  },
]
`;

exports[`select-quote-list.ts selectQuoteProps Should dispatch actions 3`] = `
Array [
  Object {
    "payload": Object {
      "clickType": "DOWN_VOTE",
      "quoteId": "a936264fd5671771d4033893764022d2",
    },
    "type": "QUOTES_TABLE::CLICK",
  },
]
`;
